<launch>
    <arg name="bag_name" default="cracker_1"/>
    <arg name="obj_name" default="cracker"/>
    <arg name="bag_full_path" default="$(find apriltag_truth)/bags/$(arg bag_name).bag"/>
    <!-- <arg name="raw_output_path" default="$(find apriltag_truth)/output/raw/"/> -->
    <arg name="filtered_output_path" default="$(find apriltag_truth)/output/filtered/"/>

    <arg name="truth_node_ns" default="truth_node" />
    <rosparam param="obj_name" subst_value="True" ns="$(arg truth_node_ns)">$(arg obj_name)</rosparam>
    <rosparam param="bag_name" subst_value="True" ns="$(arg truth_node_ns)">$(arg bag_name)</rosparam>
    <!-- <rosparam param="raw_output_path" subst_value="True">$(arg raw_output_path)</rosparam> -->
    <rosparam param="filtered_output_path" subst_value="True" ns="$(arg truth_node_ns)">$(arg filtered_output_path)</rosparam>
    <rosparam command="load" file="$(find apriltag_truth)/config/object_settings.yaml" ns="$(arg truth_node_ns)" />


    <node pkg="rosbag" type="play" name="player"  args="--delay=5 --queue=20 -r 0.5 -k  $(arg bag_full_path)"/>
    <node pkg="apriltag_truth" type="truth.py" output="screen" name="$(arg truth_node_ns)" ns="$(arg truth_node_ns)" required="true"/>

    <arg name="apriltag_node_ns" default="apriltag_ros_continuous_node" />
    <arg name="camera_name" default="/zed2/zed_node/left" />
    <arg name="image_topic" default="uncompressed"/>
    <node pkg="image_transport" type="republish" name="image_republisher0" required="true" args="compressed in:=$(arg camera_name)/image_rect_color raw out:=$(arg camera_name)/$(arg image_topic) _image_transport:=compressed" />
    <rosparam command="load" file="$(find apriltag_truth)/config/apriltag_settings.yaml" ns="$(arg apriltag_node_ns)" />
    <rosparam command="load" file="$(find apriltag_truth)/config/tags.yaml" ns="$(arg apriltag_node_ns)" />
    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg apriltag_node_ns)" clear_params="true" output="screen" >
        <!-- Remap topics from those used in code to those on the ROS network -->
        <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
        <remap from="camera_info" to="$(arg camera_name)/camera_info" />

        <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
    </node>
</launch>
